### ISSUE TITLE
Apps crash on startup libc : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR) after QPR2 merge

### ISSUE THREAD

**Issue by wodanesdag** [crdroidandroid/issue_tracker#374]
### Issue type

Source code related

### Device

Redmi Note 8 Pro (begonia)

### crDroid version

crDroid 10

### Exact version / Build date

crDroidAndroid-14.0-20240412-begonia-v10.4

### Bug description

Apps crash on startup
src.com.bni

### Steps to reproduce

Install app, start app

### Relevant log (logcat/build log)

```shell
--------- beginning of crash
04-15 14:03:32.468  4554  4554 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 4554 (src.com.bni), pid 4554 (src.com.bni)
04-15 14:03:32.489  1362  1479 W UserManagerService: Requested status bar icon for non-badged user 0
04-15 14:03:32.686  1362  3191 W ProcessStats: Tracking association SourceState{51db2f1 com.google.android.gms.persistent/10359 Fgs #1553} whose proc state 3 is better than process ProcessState{bca087 com.google.android.gms/10359 pkg=com.google.android.gms} proc state 6 (11838 skipped)
04-15 14:03:32.785  1911  1984 V WindowManagerShell: Transition animation finished (aborted=false), notifying core (#10) android.os.BinderProxy@f4d0025@0
04-15 14:03:32.788  1362  1389 V WindowManager: Finish Transition (#10): created at 04-15 14:03:32.272 collect-started=0.017ms request-sent=14.073ms started=28.477ms ready=162.505ms sent=167.933ms finished=514.237ms
04-15 14:03:32.790  1911  1984 V WindowManagerShell: Track 0 became idle
04-15 14:03:32.790  1911  1984 V WindowManagerShell: All active transition animations finished
04-15 14:03:32.816  1362  1878 W UserManagerService: Requested status bar icon for non-badged user 0
04-15 14:03:32.817  1362  1878 W UserManagerService: Requested status bar icon for non-badged user 0
04-15 14:03:32.834  1362  1521 D CompatibilityChangeReporter: Compat change id reported: 161252188; UID 10326; state: DISABLED
04-15 14:03:32.839  1362  3191 D CompatibilityChangeReporter: Compat change id reported: 253665015; UID 10326; state: ENABLED
04-15 14:03:32.839  1362  3191 D CompatibilityChangeReporter: Compat change id reported: 194532703; UID 10326; state: ENABLED
04-15 14:03:32.839  1362  3191 D CompatibilityChangeReporter: Compat change id reported: 311402873; UID 10326; state: DISABLED
04-15 14:03:32.871  4588  4588 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
04-15 14:03:32.871  4588  4588 F DEBUG   : crDroid Version: '10.4'
04-15 14:03:32.871  4588  4588 F DEBUG   : Build fingerprint: 'Redmi/begonia/begonia:11/RP1A.200720.011/V12.5.5.0.RGGMIXM:user/release-keys'
04-15 14:03:32.871  4588  4588 F DEBUG   : Revision: '0'
04-15 14:03:32.871  4588  4588 F DEBUG   : ABI: 'arm64'
04-15 14:03:32.871  4588  4588 F DEBUG   : Timestamp: 2024-04-15 14:03:32.644486773+0700
04-15 14:03:32.871  4588  4588 F DEBUG   : Process uptime: 2s
04-15 14:03:32.871  4588  4588 F DEBUG   : Cmdline: src.com.bni
04-15 14:03:32.871  4588  4588 F DEBUG   : pid: 4554, tid: 4554, name: src.com.bni  >>> src.com.bni <<<
04-15 14:03:32.871  4588  4588 F DEBUG   : uid: 10433
04-15 14:03:32.871  4588  4588 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0000000000000000
04-15 14:03:32.871  4588  4588 F DEBUG   : Cause: null pointer dereference
04-15 14:03:32.871  4588  4588 F DEBUG   :     x0  0000000000000000  x1  0000000000000000  x2  f92d448f39db55ba  x3  0000000000000000
04-15 14:03:32.871  4588  4588 F DEBUG   :     x4  0000000000000000  x5  0000000000000000  x6  0000000000000006  x7  000000717850a4f8
04-15 14:03:32.871  4588  4588 F DEBUG   :     x8  0000000000000008  x9  0000000003882cf6  x10 0000000000000018  x11 000000000000bf6a
04-15 14:03:32.871  4588  4588 F DEBUG   :     x12 0000000000000001  x13 0000000000000000  x14 0000000000000004  x15 ffffffffffffffff
04-15 14:03:32.871  4588  4588 F DEBUG   :     x16 00000071789c9280  x17 00000074a00cb32c  x18 00000074a7ca4000  x19 0000007fe3bb1780
04-15 14:03:32.871  4588  4588 F DEBUG   :     x20 0000007fe3bb17a8  x21 00000071789dc000  x22 00000071789db000  x23 00000071789da000
04-15 14:03:32.871  4588  4588 F DEBUG   :     x24 0000007fe3bb18f8  x25 00000071789c9000  x26 00000074a8ba0000  x27 0000000000000003
04-15 14:03:32.871  4588  4588 F DEBUG   :     x28 0000000000000000  x29 0000007fe3bb1730
04-15 14:03:32.871  4588  4588 F DEBUG   :     lr  00000071787b3e70  sp  0000007fe3bb1730  pc  00000071787b3e80  pst 0000000060001000
04-15 14:03:32.871  4588  4588 F DEBUG   : 47 total frames
04-15 14:03:32.871  4588  4588 F DEBUG   : backtrace:
04-15 14:03:32.871  4588  4588 F DEBUG   :       #00 pc 0000000000077e80  /data/app/~~xvP0rZiMzmZrg3Zw5wg39A==/src.com.bni--LE1zisAexepkZ7ECJW7Dw==/lib/arm64/libewxxwdf.so
04-15 14:03:32.871  4588  4588 F DEBUG   :       #01 pc 0000000000077f84  /data/app/~~xvP0rZiMzmZrg3Zw5wg39A==/src.com.bni--LE1zisAexepkZ7ECJW7Dw==/lib/arm64/libewxxwdf.so
04-15 14:03:32.871  4588  4588 F DEBUG   :       #02 pc 000000000006aeb4  /apex/com.android.runtime/bin/linker64 (__dl__ZN6soinfo17call_constructorsEv+924) (BuildId: 17b3079752491084c1a9fc88c0eae35d)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #03 pc 0000000000055ef8  /apex/com.android.runtime/bin/linker64 (__dl__Z9do_dlopenPKciPK17android_dlextinfoPKv+1040) (BuildId: 17b3079752491084c1a9fc88c0eae35d)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #04 pc 0000000000055088  /apex/com.android.runtime/bin/linker64 (__dl__ZL10dlopen_extPKciPK17android_dlextinfoPKv.__uniq.234527301065430621646263515731762262959+72) (BuildId: 17b3079752491084c1a9fc88c0eae35d)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #05 pc 00000000000010c8  /apex/com.android.runtime/lib64/bionic/libdl.so (android_dlopen_ext+16) (BuildId: 2419dcb6c791c7779c51a49aece0a721)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #06 pc 00000000000182b8  /apex/com.android.art/lib64/libnativeloader.so (android::NativeLoaderNamespace::Load(char const*) const+140) (BuildId: ee0885f563c92ea36ee55434a3827e83)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #07 pc 000000000000953c  /apex/com.android.art/lib64/libnativeloader.so (OpenNativeLibrary+140) (BuildId: ee0885f563c92ea36ee55434a3827e83)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #08 pc 000000000067ac80  /apex/com.android.art/lib64/libart.so (art::JavaVMExt::LoadNativeLibrary(_JNIEnv*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, _jobject*, _jclass*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*)+744) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #09 pc 000000000000613c  /apex/com.android.art/lib64/libopenjdkjvm.so (JVM_NativeLoad+364) (BuildId: 82c674b41deb210f0289d05447feabc2)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #10 pc 00000000000a493c  /system/framework/arm64/boot.oat (art_jni_trampoline+156) (BuildId: e4c04f812b73f711ebfe4ce93f06ec09659a94d0)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #11 pc 00000000000b4b24  /system/framework/arm64/boot.oat (java.lang.Runtime.loadLibrary0+308) (BuildId: e4c04f812b73f711ebfe4ce93f06ec09659a94d0)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #12 pc 00000000000b5d5c  /system/framework/arm64/boot.oat (java.lang.Runtime.loadLibrary0+428) (BuildId: e4c04f812b73f711ebfe4ce93f06ec09659a94d0)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #13 pc 00000000000b97cc  /system/framework/arm64/boot.oat (java.lang.System.loadLibrary+92) (BuildId: e4c04f812b73f711ebfe4ce93f06ec09659a94d0)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #14 pc 00000000003cee40  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+640) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #15 pc 0000000000613a98  /apex/com.android.art/lib64/libart.so (bool art::interpreter::DoCall<false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, bool, art::JValue*)+1820) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #16 pc 00000000006522ec  /apex/com.android.art/lib64/libart.so (void art::interpreter::ExecuteSwitchImplCpp<false>(art::interpreter::SwitchImplContext*)+14008) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #17 pc 00000000003e79d8  /apex/com.android.art/lib64/libart.so (ExecuteSwitchImplAsm+8) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #18 pc 0000000000485db8  /data/app/~~xvP0rZiMzmZrg3Zw5wg39A==/src.com.bni--LE1zisAexepkZ7ECJW7Dw==/oat/arm64/base.vdex (com.liapp.x.ݱڭֳݴ߰+0)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #19 pc 00000000003f93d8  /apex/com.android.art/lib64/libart.so (art::interpreter::ExecuteSwitch(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool) (.__uniq.112435418011751916792819755956732575238.llvm.1442440309229545803)+64) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #20 pc 00000000003f8e4c  /apex/com.android.art/lib64/libart.so (art::interpreter::Execute(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame&, art::JValue, bool, bool) (.__uniq.112435418011751916792819755956732575238.llvm.1442440309229545803)+416) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #21 pc 00000000003f8630  /apex/com.android.art/lib64/libart.so (artQuickToInterpreterBridge+776) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #22 pc 00000000003e5498  /apex/com.android.art/lib64/libart.so (art_quick_to_interpreter_bridge+88) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #23 pc 000000000072c958  /apex/com.android.art/lib64/libart.so (nterp_helper+152) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #24 pc 0000000000611de4  /data/app/~~xvP0rZiMzmZrg3Zw5wg39A==/src.com.bni--LE1zisAexepkZ7ECJW7Dw==/oat/arm64/base.vdex (src.com.bni.MoEngageApplication.attachBaseContext+0)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #25 pc 000000000023f638  /system/framework/arm64/boot-framework.oat (android.app.Instrumentation.newApplication+152) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #26 pc 0000000000247510  /system/framework/arm64/boot-framework.oat (android.app.LoadedApk.makeApplicationInner+1968) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #27 pc 000000000030e174  /system/framework/arm64/boot-framework.oat (android.app.ActivityThread.handleBindApplication+5524) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #28 pc 000000000030795c  /system/framework/arm64/boot-framework.oat (android.app.ActivityThread$H.handleMessage+8732) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #29 pc 0000000000576408  /system/framework/arm64/boot-framework.oat (android.os.Handler.dispatchMessage+168) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #30 pc 00000000005795b4  /system/framework/arm64/boot-framework.oat (android.os.Looper.loopOnce+1140) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #31 pc 000000000057908c  /system/framework/arm64/boot-framework.oat (android.os.Looper.loop+252) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #32 pc 00000000003180c4  /system/framework/arm64/boot-framework.oat (android.app.ActivityThread.main+1700) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #33 pc 00000000003cee40  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+640) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #34 pc 00000000003c5370  /apex/com.android.art/lib64/libart.so (_jobject* art::InvokeMethod<(art::PointerSize)8>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+888) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #35 pc 00000000003c4fe8  /apex/com.android.art/lib64/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*) (.__uniq.165753521025965369065708152063621506277)+32) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #36 pc 00000000000aa4c8  /system/framework/arm64/boot.oat (art_jni_trampoline+120) (BuildId: e4c04f812b73f711ebfe4ce93f06ec09659a94d0)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #37 pc 000000000075cf24  /system/framework/arm64/boot-framework.oat (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+132) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.871  4588  4588 F DEBUG   :       #38 pc 0000000000766c88  /system/framework/arm64/boot-framework.oat (com.android.internal.os.ZygoteInit.main+3784) (BuildId: 92543253b03e226c5bda7ea068a6332252e06f42)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #39 pc 00000000003cee40  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+640) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #40 pc 0000000000387c14  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+220) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #41 pc 00000000005cf824  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeWithVarArgs<_jmethodID*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+320) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #42 pc 00000000006f4d7c  /apex/com.android.art/lib64/libart.so (art::JNI<true>::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+152) (BuildId: 00954f1580a31523e8571381665e0b94)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #43 pc 00000000000d7640  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+104) (BuildId: f11a8f4b92f74697ed3b8d71b4b7e518)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #44 pc 00000000000e3260  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+836) (BuildId: f11a8f4b92f74697ed3b8d71b4b7e518)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #45 pc 0000000000002580  /system/bin/app_process64 (main+1224) (BuildId: af111bda59ddc4b8b33df988040c43b6)
04-15 14:03:32.872  4588  4588 F DEBUG   :       #46 pc 00000000000544c4  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+104) (BuildId: a5b98cf42d1d47dc94c5d682c0f8ed4c)
04-15 14:03:32.893  1362  4629 I DropBoxManagerService: add tag=data_app_native_crash isTagEnabled=true flags=0x2
04-15 14:03:32.903  1362  1511 I BootReceiver: Copying /data/tombstones/tombstone_00 to DropBox (SYSTEM_TOMBSTONE)
04-15 14:03:32.903  1362  1511 I DropBoxManagerService: add tag=SYSTEM_TOMBSTONE isTagEnabled=true flags=0x6
04-15 14:03:32.914  1362  1362 W AccessibilityManagerService:  wait for adding window timeout: 9
04-15 14:03:32.924  1362  1511 I DropBoxManagerService: add tag=SYSTEM_TOMBSTONE_PROTO_WITH_HEADERS isTagEnabled=true flags=0x4
04-15 14:03:32.961  1362  2974 I ActivityManager: Process src.com.bni (pid 4554) has died: fg  TOP 
04-15 14:03:32.962  1362  1390 D DisplayManagerService: Drop pending events for gone uid 10433
04-15 14:03:33.173  1362  1508 D CompatibilityChangeReporter: Compat change id reported: 218533173; UID 10425; state: ENABLED
04-15 14:03:33.173  1362  1508 D CompatibilityChangeReporter: Compat change id reported: 262645982; UID 10425; state: ENABLED
04-15 14:03:33.179  1362  1508 D CompatibilityChangeReporter: Compat change id reported: 218533173; UID 10415; state: ENABLED
04-15 14:03:33.179  1362  1508 D CompatibilityChangeReporter: Compat change id reported: 262645982; UID 10415; state: ENABLED
04-15 14:03:33.183  1362  1508 D CompatibilityChangeReporter: Compat change id reported: 218533173; UID 10421; state: ENABLED
04-15 14:03:33.183  1362  1508 D CompatibilityChangeReporter: Compat change id reported: 262645982; UID 10421; state: ENABLED
04-15 14:03:33.191  1362  1399 W VoiceInteractionManager: no available voice recognition services found for user 0
04-15 14:03:33.247  1362  1399 W VoiceInteractionManager: no available voice recognition services found for user 0
04-15 14:03:33.276  1362  1399 W VoiceInteractionManager: no available voice recognition services found for user 0
04-15 14:03:33.288  1362  1399 W VoiceInteractionManager: no available voice recognition services found for user 0
04-15 14:03:33.340  1362  1399 W VoiceInteractionManager: no available voice recognition services found for user 0
04-15 14:03:33.347  1362  1399 W VoiceInteractionManager: no available voice recognition services found for user 0
04-15 14:03:33.353  1362  1399 W VoiceInteractionManager: no available voice recognition services found for user 0
04-15 14:03:33.376  1362  1399 W GameServiceProviderSelector: No available game service found for user id: 0
04-15 14:03:33.643  1362  1477 D OomAdjuster: Not killing cached processes
```

### Additional context

Also happened on my personal unofficial build for Redmi Note 12 Pro 5G (ruby)

### Acknowledgements

- [X] I've checked device is officially supported (for device specific reports and not source related).
- [X] I'm running latest version available on crdroid.net for this device
- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by gwolf2u** [comment]
did you install proper gapps?

**Comment by wodanesdag** [comment]
yes, it worked on 10.2 then stopped working after update to 10.4 through updater
clean flash 10.4 + gapps not working
clean flash 10.2 + gapps working
i've tried other roms that has merged QPR2 (Matrixx-v10.4.1, Project Elixir v4.2) and it works there so probably not QPR2 specific but the commits after that

**Comment by neobuddy89** [comment]
This is fixed in latest source.
